circuits/src/zk_gadgets/select.rs

```diff
@@ -96,58 +36,26 @@ impl MultiproverCondSelectGadget {
 pub struct CondSelectVectorGadget {}
 impl CondSelectVectorGadget {
     /// Implements the control flow statement if selector { a } else { b }
-    pub fn select<L1, L2, V1, V2, CS>(a: &[V1], b: &[V1], selector: L1, cs: &mut CS) -> Vec<V2>
-    where
-        L1: LinearCombinationLike,
-        L2: LinearCombinationLike,
-        V1: CircuitVarType<L2>,
-        V2: CircuitVarType<LinearCombination>,
-        CS: RandomizableConstraintSystem,
-    {
-        assert_eq!(a.len(), b.len(), "a and b must be of equal length");
-
-        let mut selected = Vec::with_capacity(a.len());
-        for (a_val, b_val) in a.iter().zip(b.iter()) {
-            selected.push(CondSelectGadget::select(
-                a_val.clone(),
-                b_val.clone(),
-                selector.clone(),
-                cs,
-            ));
-        }
-
-        selected
-    }
-}
-
-/// A multiprover variant of the CondSelectVectorGadget
-pub struct MultiproverCondSelectVectorGadget;
```

Comment:
just making sure my understanding is correct: we don't need a `MultiProverCondSelectVectorGadget` anymore because the implementation differences b/t single/multiprover for the mux gate are handled by the constraint system, over which the gadget is generic, right?

if so, is the general high-level goal to have all gadgets be implemented in a way that is agnostic of a single/multiprover setting, and have this abstracted entirely by the choice of constraint system?

Reply:
That is correct, the `mux` implementation defines the differences. I'd say this is _generally_ the goal, but it won't be achieved for all gadgets. Some are really hard to do in an MPC so it's not worth implementing just for generic functionality. 

But generally yes, note that those already done (poseidon, bits, etc) are all generic over the proof system